Sources/PackageModel/Manifest/PackageDependencyDescription.swift

```diff
@@ -11,15 +11,33 @@
 import TSCBasic
 
 /// Represents a package dependency.
-public struct PackageDependencyDescription: Equatable, Codable, Hashable {
+public enum PackageDependencyDescription: Equatable {
```

Comment:
👀 this is the other key change PackageDependencyDescription is now and enum with two cases: local and git. the idea is to model the data more accurately instead of the a struct that has fields that are reused with different meaning and make reasoning in downstream harder

Reply:
The one drawback (since enums can't have stored properties) is that common properties have to be replicated across the payloads.  I don't think this is a problem but another way that might make sense would be to separate out the reference type as an enum (path, git repo, registry), which would also make clear what is common to all packages (e.g. identity and filter) but leaving the specifics of the reference type to the enum.  Just a thought.

Reply:
yes, this is the tradeoff here. i went with creating computed properties for the shared properties since its not a big price to pay, and the price is paid only once in PackageDependencyDescription itself, keeping the call-sites nice and clean. if we split the properties across the associated values and and some "type" enum, the call-sites would have to know how to access the properties so the code in PackageDependencyDescription would be cleaner but the call-sites "uglier"

Reply:
Agreed.  Overall it's a good tradeoff.

Reply:
my thoughts exactly